### DISCUSSION TITLE
Is Vercel suitable to host Meilisearch

### DISCUSSION THREAD

**Discussion by mrmartineau** [vercel/community#454] [Help] [answered]
I would like to run an instance of [Meilisearch](https://www.meilisearch.com/) but I am not sure if Vercel's hosting model is appropriate for it. It is essentially a self-hosted clone of Algolia. 

Would the search index or any preferences get wiped when a new deployment occurs? Am I thinking about this in the wrong way?

**Comment by paulogdm** [comment, marked as answer]
Hello,

We recommend that you use serverful providers for API-only deployments. While Serverless Functions can be a good fit, heavy usage of those resources can breach the [Fair Use Policy](https://vercel.com/docs/platform/fair-use-policy) pretty quickly.

**Reply by mrmartineau** [reply]
Thanks @paulogdm I didn't realise that would be an issue. I will try another provider.

My question was more about if the search index/preferences would be wiped each time a deployment occurred. Can you speak to that?

**Reply by paulogdm** [reply]
Yes, it would. But in a Serverless Environment, all deployments are "stateless" and you can lose state in a few minutes. If you need data-retention, a DB should be used, for example.

**Reply by mrmartineau** [reply]
ok, cool. That's what I thought. Thanks

**Reply by ThatGuySam** [reply]
If you were to combine a CDN like Cloudflare with a serverless Meilisearch solution like [this](https://www.christopherbiscardi.com/running-meilisearch-in-a-lambda) would that enable using Meilisearch without breaching the Fair Use Policy?
